konduit-serving-core/src/main/java/ai/konduit/serving/executioner/PipelineExecutioner.java

```diff
@@ -209,6 +212,9 @@ public void init() {
         ServingConfig servingConfig = config.getServingConfig();
         //initialize input and output data types
         this.pipeline = Pipeline.getPipeline(config.getSteps());
+        Preconditions.checkState(config.getSteps().get(0).isValidInputType(servingConfig.getInputDataFormat()),"Configured Serving Configuration input type is invalid");
```

Comment:
Can we log the configured type and tell the user which ones are allowed (keeping in mind null means any)? Otherwise this might be difficult to debug, esp. for Python users.